### PR TITLE
VDPN-1/VDPN-25 BE - Create DB model

### PR DESCRIPTION
**Pull request by Alina-Aurica**
1. I removed the fields: visitHour, visitMonth, and numberOfVisits from VisitStatistics, as I believe these should be derived through a service that analyzes all the records in this table.
2. I included an isLogged field in the User model to enhance the notification feature.

### CODE REVIEW

File: TravelPoints-BackEnd/pom.xml [hunk 1]
```diff
@@ -45,7 +45,22 @@
...
+        </dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-validation</artifactId>
+		</dependency>
+		<dependency>
+            <groupId>jakarta.validation</groupId>
+            <artifactId>jakarta.validation-api</artifactId>
+            <version>3.0.2</version>
+        </dependency>
```
Comment (iDietzer) [inline TravelPoints-BackEnd/pom.xml:62]: C: make sure everything in indented correctly

File: TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/AttractionDetail.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    private static final long serialVersionUID = 3L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long detailId;
+    private Long attractionId;
+    private String descriptionText;
+    private byte[] descriptionAudio; // or Blob
+    private Float entryPrice;
+    private String offers;
```
Comment (iDietzer) [inline TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/AttractionDetail.java]: M: use `@Column(<<column_name>>)` for every field (See here: https://www.geeksforgeeks.org/spring-data-jpa-column-annotation/?ref=lbp)

File: TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/Review.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private static final long serialVersionUID = 1L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long reviewId;
+    private Long userId;
+    private Long attractionId;
+    private String reviewText;
+    private Integer rating;
+    private Timestamp createdAt;
```
Comment (iDietzer) [inline TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/Review.java]: M: see above, also use mappings for relations between tables. (See here: https://stackabuse.com/a-guide-to-jpa-with-hibernate-relationship-mapping/)

File: TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/TouristAttraction.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+public class TouristAttraction implements Serializable {
+    @Serial
+    private static final long serialVersionUID = 2L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long attractionId;
+    private String location;
+    private String category;
+    private Timestamp createdAt;
```
Comment (iDietzer) [inline TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/TouristAttraction.java]: M: see above

File: TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/User.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long userId;
+    @Column(unique = true)
+    private String email;
+    private String password;
+    @Enumerated(EnumType.STRING)
+    private UserType userType;
+    private Timestamp createdAt;
+    private boolean isLogged;
```
Comment (iDietzer) [inline TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/User.java]: M: see above

File: TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/UserWishlist.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+public class UserWishlist implements Serializable {
+    @Serial
+    private static final long serialVersionUID = 4L;
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long wishlistId;
+    private Long userId;
+    private Long attractionID;
+    private Timestamp addedAt;
+
```
Comment (iDietzer) [inline TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/model/UserWishlist.java]: M: see above

**Review by iDietzer — CHANGES_REQUESTED** [review]

File: TravelPoints-BackEnd/pom.xml (hunk 1 above)
Comment (Alina-Aurica) [reply TravelPoints-BackEnd/pom.xml:62]: On my PC looks good
image

**Review by iDietzer — APPROVED** [review]

**Pull request merged**